lib/mayaUsd/fileio/utils/meshWriteUtils.h

```diff
@@ -82,12 +88,47 @@ namespace UsdMayaMeshUtil
     MAYAUSD_CORE_PUBLIC
     void writeInvisibleFacesData(const MFnMesh&, UsdGeomMesh&, UsdUtilsSparseValueWriter&);
 
+    MAYAUSD_CORE_PUBLIC
+    void writeNormalsData(const MFnMesh&, UsdGeomMesh&, const UsdTimeCode&, UsdUtilsSparseValueWriter&);
+
     MAYAUSD_CORE_PUBLIC
     bool getMeshUVSetData(const MFnMesh&, const MString&, VtArray<GfVec2f>*, TfToken*, VtArray<int>*);
 
     MAYAUSD_CORE_PUBLIC
     bool writeUVSetsAsVec2fPrimvars(const MFnMesh&, UsdGeomMesh&, const UsdTimeCode&, UsdUtilsSparseValueWriter&);
 
+    MAYAUSD_CORE_PUBLIC
+    bool addDisplayPrimvars(UsdGeomGprim&, const UsdTimeCode&, const MFnMesh::MColorRepresentation,
+                            const VtArray<GfVec3f>&, const VtArray<float>&, const TfToken&,
+                            const VtArray<int>&, const bool, const bool, UsdUtilsSparseValueWriter&);
+
+    MAYAUSD_CORE_PUBLIC
+    bool createRGBPrimVar(UsdGeomGprim&, const TfToken&, const UsdTimeCode&,
+                          const VtArray<GfVec3f>&, const TfToken&, const VtArray<int>&,
+                          bool, UsdUtilsSparseValueWriter&);
+
+    MAYAUSD_CORE_PUBLIC
+    bool createRGBAPrimVar(UsdGeomGprim&, const TfToken&, const UsdTimeCode&,
+                           const VtArray<GfVec3f>&, const VtArray<float>&, 
+                           const TfToken&,const VtArray<int>&,bool, UsdUtilsSparseValueWriter&);
+
+    MAYAUSD_CORE_PUBLIC
+    bool createAlphaPrimVar(UsdGeomGprim&, const TfToken&, const UsdTimeCode&,
+                            const VtArray<float>&, const TfToken&,const VtArray<int>&,
+                            bool, UsdUtilsSparseValueWriter&);
+
+    /// Collect values from the color set named \p colorSet.
+    /// If \p isDisplayColor is true and this color set represents displayColor,
+    /// the unauthored/unpainted values in the color set will be filled in using
+    /// the shader values in \p shadersRGBData and \p shadersAlphaData if available.
+    /// Values are gathered per face vertex, but then the data is compressed to
+    /// vertex, uniform, or constant interpolation if possible.
+    /// Unauthored/unpainted values will be given the index -1.
+    MAYAUSD_CORE_PUBLIC
+    bool getMeshColorSetData( MFnMesh&, const MString&, bool, const VtArray<GfVec3f>&, const VtArray<float>&,
+                              const VtArray<int>&, VtArray<GfVec3f>*, VtArray<float>*, TfToken*, VtArray<int>*,
+                              MFnMesh::MColorRepresentation*, bool*);
+
```

Comment:
These changes dates back a while when I was just focusing on creating the utility functions without paying attention to argument names in the header. At the time, I was also making changes to reduce the number of passed parameters but eventually abandoned the work because it was a lot more work for the sake of PR size. That's why I probably have missed to put back the parameters name. My bad!

Speaking of parameters, a lot of these utility functions takes more than 6 or 7 arguments. There is even one that takes 12 arguments :). One way would be to group some of these parameters in a structure and pass it around.  Something to think about...

See 35c0c0fd6964f2970a512397546036a73af201df